lab_2_tokenize_by_bpe/main.py

```diff
@@ -112,25 +112,40 @@ def train(
         if num_merges > len(pairs_of_tokens):
             num_merges = len(pairs_of_tokens)
         max_prevalence = max(pairs_of_tokens.values())
-        max_prev_pairs = []
-        for pair, frequency in word_frequencies.items():
-            if frequency == max_prevalence:
-                max_prev_pairs.append(pair)
+        max_prev_pairs = [pair for pair in pairs_of_tokens if pairs_of_tokens[pair] == max_prevalence]
+        max_length = max([len(str(pair)) for pair in max_prev_pairs])
+        max_length_pairs = [pair for pair in max_prev_pairs if len(str(pair)) == max_length]
+        result = sorted(max_length_pairs)
```

Comment:
Good, but what if you used all those conditions you've filtered by as a key argument to `sorted`?